Add `TF_STATE_PERSIST_INTERVAL` environment variable. (#35176)
hashicorp/terraform: website/docs/cli/config/environment-variables.mdx

export TF_REGISTRY_CLIENT_TIMEOUT=15
```

## TF_STATE_PERSIST_INTERVAL

> The interval in seconds Terraform will attempt to persist state to a remote backend during an apply operation. The default minimum interval for all remote backends is 20 seconds, however certain backends may choose to override the default value. If the value of this interval is lower than the default interval, the default will be used.

Reviewer: Small style suggestions for active voice and present tense:
```suggestion
The interval in seconds that Terraform attempts to persist state to a remote backend during an apply operation. The default minimum interval for all remote backends is 20 seconds. Backends may override the default value, but Terraform uses the default if a backend's specified interval is less than the default interval.
```
